worker/src/prompt.ts

```diff
@@ -0,0 +1,69 @@
+
+export interface Prompt {
+    prompt: string;
+    values: { [key: string]: string };
+}
+ 
+export interface PromptResult {
+    request: Request,
+    body: string,
+    prompt?: Prompt,
+}
+
+function formatPrompt(prompt: Prompt): string {
+    let formattedString = prompt.prompt;
+
+    for (const key in prompt.values) {
+        const placeholder = new RegExp(`{{${key}}}`, 'g');
+        formattedString = formattedString.replace(placeholder, prompt.values[key]);
+    }
+
+    return formattedString;
+}
+
+function updateContentLength(clone: Request, text: string): Request {
+    const body = new TextEncoder().encode(text);
+    const headers = new Headers(clone.headers);
+    headers.set("Content-Length", `${body.byteLength}`);
+
+    return new Request(clone.url, {
+        method: clone.method,
+        headers,
+        body
+    });
+}
+
+export async function extractPrompt(
+    request: Request,
+): Promise<PromptResult> {
+    try {
+        const isPromptRegexOn = request.headers.get("Helicone-Prompt-Format") !== null;
+
+        if (isPromptRegexOn) {
+            const cloneRequest = request.clone();
+            const cloneBody = await cloneRequest.text();
+            const json = cloneBody ? JSON.parse(cloneBody) : {};
+            const prompt = JSON.parse(json["prompt"])
+            const stringPrompt = formatPrompt(prompt);
+            json["prompt"] = stringPrompt
+            const body = JSON.stringify(json);
+            const formattedRequest = updateContentLength(cloneRequest, body);
+
+            return {
+                request: formattedRequest,
+                body: body,
+                prompt: prompt,
+            }
+        } else {
+            return {
+                request: request,
+                body: await request.text(),
+            }
+        }
+    } catch (error) {
+        return {
+            request: request,
+            body: await request.text(),
+        }
+    }
```

Comment:
I think we want to change the logic a bit here, so that we don't fall back and just pass the request through. If we fail to `formatPrompt` we should error and report that error
```suggestion

      const isPromptRegexOn = request.headers.get("Helicone-Prompt-Format") !== null;

  if (isPromptRegexOn) {
    try {
      const cloneRequest = request.clone();
      const cloneBody = await cloneRequest.text();
      const json = cloneBody ? JSON.parse(cloneBody) : {};
      const prompt = JSON.parse(json["prompt"]);
      const stringPrompt = formatPrompt(prompt);
      json["prompt"] = stringPrompt;
      const body = JSON.stringify(json);
      const formattedRequest = updateContentLength(cloneRequest, body);

      return {
        request: formattedRequest,
        body: body,
        prompt: prompt,
      };
    } catch (error) {
      //TODO HANDLE ERROR BETTER
    }
  } else {
    return {
      request: request,
      body: await request.text(),
    };
  }

```

Reply:
great call, agreed